### ISSUE TITLE
Grouped buttons

### ISSUE THREAD

**Issue by matteoferigo** [felixerdy/touchbar#16]
Hi, I saw that there is a type option named group, but I can't see any documentation about that.
How can I implement them?

Thanks

**Comment by Rawnly** [comment]
I think that is the same as the [electron touchbar documentation](https://electron.atom.io/docs/api/touch-bar/)

**Comment by matteoferigo** [comment]
I think so, but the result is just a list of single buttons (not grouped).

```
{
  "type": "group",
  "elements": [{
    "name": "add-file",
    "type": "button",
    "label": "Add file",
    "command": "tree-view:add-file"
  }, {
    "name": "add-folder",
    "type": "button",
    "label": "Add folder",
    "command": "tree-view:add-folder"
    }]
}
```

**Comment by Rawnly** [comment]
Yes, I think that the "group" as you mean is not supported by Electron Touchbar API

**Comment by felixerdy** [comment]
Maybe you are looking for a [Segmented Control](https://github.com/electron/electron/blob/master/docs/api/touch-bar-segmented-control.md)? This function is currently not included in this package...

**Comment by Rawnly** [comment]
> Maybe you are looking for a Segmented Control? This function is currently not included in this package...

I don't think so, this is the segment control:
<img width="1085" alt="touch bar shot 2017-10-25 at 15 20 30" src="https://user-images.githubusercontent.com/16429579/32000763-0ddbbeea-b998-11e7-8847-12ea1cdbc2a9.png">

and I think he means something like this:

> Group label
<img width="1085" alt="touch bar shot 2017-10-25 at 15 22 50" src="https://user-images.githubusercontent.com/16429579/32000883-66d7ec08-b998-11e7-96be-9bc41a9844ee.png">

> Group Items
<img width="1085" alt="touch bar shot 2017-10-25 at 15 22 52" src="https://user-images.githubusercontent.com/16429579/32000888-6b57d270-b998-11e7-857a-9f217b8142dd.png">

The screenshots above are from Better Touch Tool

**Comment by matteoferigo** [comment]
Whoops! I meant the segment control, just to group a set of buttons.
Is it planned to be released in a next release?

**Comment by felixerdy** [comment]
Ah okay, he could use the Popover which is kind of similar

**Comment by Rawnly** [comment]
> Ah okay, he could use the Popover which is kind of similar

Yes I think so, I'm testing them all and it's seems to be the right one!

**Comment by Rawnly** [comment]
@felixerdy it's could be cool to have an extra file to edit, with all the touchbar items instead the settings textarea.

It could be more productive, and support syntax highlighting, linters etc.. 

Just a simple javascript module


```js
  module.exports = [
  //  ...components
]
```

**Comment by matteoferigo** [comment]
@felixerdy I would like to group some buttons to get a quicker accessibility 
Example: [Add file|Add folder] [Beautify]

**Comment by idoo** [comment]
any news?

**Comment by matteoferigo** [comment]
I created a [PR to introduce touch bar segmented control](https://github.com/felixerdy/touchbar/pull/38)
Unfortunately, this will not close this issue, because the aspected behavior is a group of buttons.
What [segmented control does] 
(https://electronjs.org/docs/api/touch-bar-segmented-control) is simply to show a bunch of [selectable segments](https://electronjs.org/docs/api/structures/segmented-control-segment), which have few options, nothing comparable with buttons
Still dreaming on…

**Comment by thirstyice** [comment]
See my PR #46 
I think it does what you need to do
